frontend/src/main/java/com/google/tripmeout/frontend/serialization/GsonTripModelTypeAdapter.java

```diff
@@ -0,0 +1,82 @@
+package com.google.tripmeout.frontend.serialization;
+
+import com.google.appengine.repackaged.com.google.api.client.json.Json;
+import com.google.common.base.Strings;
+import com.google.gson.JsonParseException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+import com.google.tripmeout.frontend.TripModel;
+import java.io.IOException;
+
+public class GsonTripModelTypeAdapter extends TypeAdapter<TripModel> {
+  private static final String TRIP_MODEL_ID_JSON_FIELD_NAME = "id";
+  private static final String TRIP_MODEL_NAME_JSON_FIELD_NAME = "name";
+  private static final String TRIP_MODEL_USER_ID_JSON_FIELD_NAME = "userId";
+  private static final String TRIP_MODEL_LATITUDE_JSON_FIELD_NAME = "locationLat";
+  private static final String TRIP_MODEL_LONGITUDE_JSON_FIELD_NAME = "locationLong";
+
+  @Override
+  public TripModel read(JsonReader reader) throws IOException {
+    if (reader.peek() == JsonToken.NULL) {
+      reader.nextNull();
+      return null;
+    }
+    TripModel.Builder tripBuilder = TripModel.builder();
+    reader.beginObject();
+    while (reader.hasNext()) {
+      JsonToken token = reader.peek();
+      if (token.equals(JsonToken.NAME)) {
```

Comment:
As in the other PR, it's likely safer to remove the conditional check and just call `reader.nextName()`.